### DISCUSSION TITLE
QOL improvement: allow integer values for `Optimizer.set_learning_rate()`

### DISCUSSION THREAD

**Discussion by maxfrei750** [mitsuba-renderer/mitsuba3#257] [Feature requests]
Sometimes, I forget to specify the learning rate with a decimal point and get
```
Exception has occurred: Exception
Optimizer.set_learning_rate(): value should be a float or a dict!
  File "/src/mitsuba3/build/python/mitsuba/python/ad/optimizers.py", line 103, in set_learning_rate
    raise Exception('Optimizer.set_learning_rate(): value should be a float or a dict!')
  File "/src/mitsuba3/build/python/mitsuba/python/ad/optimizers.py", line 21, in __init__
    self.set_learning_rate(lr)
  File "/src/mitsuba3/build/python/mitsuba/python/ad/optimizers.py", line 261, in __init__
    super().__init__(lr, params)
```

Would it result in problems, if we added

```
if isinstance(lr, int):
    lr = float(lr)
```
to https://github.com/mitsuba-renderer/mitsuba3/blob/6902109b9cab4607eaab47fa00bdd196ee458e51/src/python/python/ad/optimizers.py#L83-L103
?

**Comment by Speierers** [comment]
Thanks @maxfrei750 for reporting this. It will be fixed in the next release: [53143db](https://github.com/mitsuba-renderer/mitsuba3/commit/53143db05739b964b7a489f58dbd1bd4da87533c)
